File: src/C#/Parcial2/ActividadP2A4/EmpleadoPorComision.cs
using System;

public class EmpleadoPorComision : Empleado
{
    private double ventasBrutas;
    private double tarifaComision;

    public EmpleadoPorComision(string nombre, string apellido, string nss, double ventas, double tarifa)
        : base(nombre, apellido, nss)
    {
        EstablecerVentasBrutas(ventas);
        EstablecerTarifaComision(tarifa);
    }

    public void EstablecerTarifaComision(double tarifa) { tarifaComision = (tarifa > 0.0 && tarifa < 1.0) ? tarifa : 0.0; }
    public double ObtenerTarifaComision() { return tarifaComision; }
    public void EstablecerVentasBrutas(double ventas) { ventasBrutas = (ventas < 0.0) ? 0.0 : ventas; }
    public double ObtenerVentasBrutas() { return ventasBrutas; }

    public override double Ingresos()
    {
        return ObtenerTarifaComision() * ObtenerVentasBrutas();
    }

    public override string ToString()
    {
        return "empleado por comision: " + base.ToString() + "\nventas brutas: $" + ObtenerVentasBrutas() + "; tarifa de comision: " + ObtenerTarifaComision();
    }
}

File: src/Python/Parcial2/ActividadP2A4/empleadoporcomision.py
from empleado import Empleado

class EmpleadoPorComision(Empleado):
    def __init__(self, nombre, apellido, nss, ventas, tarifa):
        super().__init__(nombre, apellido, nss)
        self.establecerVentasBrutas(ventas)
        self.establecerTarifaComision(tarifa)

    def establecerTarifaComision(self, tarifa):
        if tarifa > 0.0 and tarifa < 1.0:
            self.tarifaComision = tarifa
        else:
            self.tarifaComision = 0.0

    def obtenerTarifaComision(self):
        return self.tarifaComision

    def establecerVentasBrutas(self, ventas):
        if ventas < 0.0:
            self.ventasBrutas = 0.0
        else:
            self.ventasBrutas = ventas

    def obtenerVentasBrutas(self):
        return self.ventasBrutas

    def ingresos(self):
        return self.obtenerTarifaComision() * self.obtenerVentasBrutas()

    def __str__(self):
        return f"empleado por comision: {super().__str__()}\nventas brutas: ${self.obtenerVentasBrutas()}; tarifa de comision: {self.obtenerTarifaComision()}"
